File: 1470 - Easy - Shuffle the array/solution.kt
//* Solution using a for loop and adding to a mutable list. */
class solution {
    fun shuffle(nums: IntArray, n: Int): IntArray {
        var r = mutableListOf<Int>()
        for (it in 0 until n) {
            r.add(nums[it])
            r.add(nums[it + n])
        }
        return r.toIntArray()
    }
}

//* Solution using list operations and collection transformations */
class solution {
    fun shuffle(nums: IntArray, n: Int): IntArray {
        val x = nums.slice(0 until n)
        val y = nums.slice(n until nums.size)   
        return x.zip(y).flatMap{ it.toList() }.toIntArray()
    }
}

File: 1470 - Easy - Shuffle the array/solution.py
# Solution using for loop and appending to an empty list.
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        ret = []
        for i in range(n):
            ret.append(nums[i])
            ret.append(nums[i + n])
        return ret

# Solution using list slicing and appending to an empty list.
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        ret = []
        x = nums[0:n]
        y = nums[n:]
        for i in range(n):
            ret.append(x[i])
            ret.append(y[i])
        return ret

# Solution using zip with a list comprehension.
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        return [x for y in zip(nums[0:n], nums[n:]) for x in y]